### ISSUE TITLE
Server side reports 0Mb/s bandwidth

### ISSUE THREAD

**Issue by SLX-WI** [esnet/iperf#293]
Server side reports 0Mb/s bandwidth from sender in plain text and json format. I used "iperf3 -s" and "iperf3 -c IP". git bisect revealed that bug was introduced by commit 582438111107ad3fce171afaee277bf594099ef5.

**Comment by bmah888** [comment]
OK I see this problem on the tip of the master branch.  Thanks for finding the bisection.  I'm not sure why this commit caused this problem (or rather...I could easily believe that it did...but I'm not sure how the change caused the bug).  One more mystery to solve!

**Comment by bmah888** [comment]
That was a fix for #251.  In trying to make sure we printed out the last interval's statistics on the server side, we ended up emitting all of the end-of-test server output before we'd exchanged statistics with the client.  That's why you see a bunch of 0 from the client.

I've been playing around with a few fixes but none of them is exactly right.  The closest I've come is to split the end-of-test server output into two parts...printing the last interval's statistics before we exchange statistics (so that when we exchange statistics, we have the complete set of interval statistics), and emitting the end-of-test summary after exchanging statistics (so we get all the right stuff from the client).  This sounds great, except that it partially re-breaks the bug in #251...in the ```--get-server-output``` case, the server side hasn't printed out its end-of-test summary before it pushes its output to the client.

(Note to people from the future:  That was to restore the original call to ```test->reporter_callback()``` in ```iperf_handle_message_server```, and a separate case for ```TEST_END``` in ```iperf_reporter_callback``` that does ```iperf_print_intermediate(test)```.  It might also have involved modification of the ```DISPLAY_RESULTS``` case.)

Basically the current structure of iperf3 didn't take all these cases into account, which makes it very difficult to get all of these printing and exchanging of results done in the right order for all the different cases.  I don't know if it can be solved without some serious code re-arrangement.

I'm going to leave this open but untarget the bug milestone.

**Comment by slavicat** [comment]
Hi,

 I am experiencing the same problem. Can you just tell me is it possible to extract information about the average throughput from the drop rate that is given in the results? In other words, is the print output of zero value for throughput the only bug? Are the results for jitter and drop rate correct?
